includes/class-sensei-lesson.php

```diff
@@ -3767,7 +3767,14 @@ public function set_quick_edit_admin_defaults( $column_name, $post_id ) {
 	 */
 	public static function single_course_lessons_classes( $classes ) {
 
-		if ( is_singular( 'course' ) ) {
+		/**
+		 * Filters if it is a single course.
+		 *
+		 * @since 3.3.0
+		 *
+		 * @param {bool} $is_single_course True if it is a single course.
+		 */
+		if ( apply_filters( 'sensei_is_single_course', false ) ) {
```

Comment:
I don't think that we should introduce a new filter for this. We should add an action to `sensei_single_course_content_inside_before` and then add a filter to `post_class` which calls `single_course_lessons_classes` inside it.

Reply:
Hum! Good idea!
I just implemented a little differently. I added the filter before loading the template and removed after that (in case to have another list after that where it should not be used).